libs/git/repository_test.go

```diff
@@ -209,7 +209,23 @@ func TestRepositoryGitConfigWhenNotARepo(t *testing.T) {
 }
 
 func TestRepositoryOriginUrlRemovesUserCreds(t *testing.T) {
-	repo := newTestRepository(t)
-	repo.addOriginUrl("https://username:token@github.com/databricks/foobar.git")
-	repo.assertOriginUrl("https://github.com/databricks/foobar.git")
+	tcases := []struct {
+		url      string
+		expected string
+	}{
+		{
+			url:      "https://username:token@github.com/databricks/foobar.git",
+			expected: "https://github.com/databricks/foobar.git",
+		},
+		{
+			url:      "https://token@github.com/databricks/foobar.git",
```

Comment:
Nit: this is interpreted as username, not token, AFAIK.

Reply:
Yeak, you are absolutely correct. However it still works. That is Github's integration with the `git` CLI allows for pat tokens to be set as the user name.

Added a comment explaining as much.